src/Endatix.Core/Entities/Submission.cs

```diff
@@ -50,6 +46,11 @@ public void Update(string jsonData, long formDefinitionId, bool isComplete = tru
         SetCompletionStatus(isComplete);
     }
 
+    public void UpdateToken(Token token)
```

Comment:
I understand this is because of DDD and encapsulation but in practice we achieve the same just not utilizing one the the oldest benefits of C#.
The properties are actually methods for getting and setting. From the early days of C# they've always made a difference with languages like e.g. Java, where one needs to add many GetXxx and SetXxx methods for something that in C# is done in a much nicer way. IMO using a method UpdateXxx that simply sets a private property that simply sets a private field makes the code less clean and tidy.
This is a general approach discussion and I believe this PR is not the place for it so let's clarify it offline.

Reply:
This is not aimed change the C# syntax sugar, but actually to use the power of the language and hide the Data internals of setters, so we can only interact with the encapsulated Domain Model.

This is part of the domain driven design recommendations to avoid Anemic Domain Model (https://martinfowler.com/bliki/AnemicDomainModel.html), so we can encapsulate the logic and force changes through one method only, where we can take care of raising domain events, running validation and working with the domain. Idea is to hide internal data and emmite domain behaviors and this is why the setters are set to private and hence the reason to fix this as all other props had private setters.

Reply:
Yes, I know this theory and the ideas, they are tens of years old. What I mean is that it all is achievable with using what C# gives us to make the code cleaner and tidier.